upup/pkg/kutil/rollingupdate_cluster_validate_drain.go

```diff
@@ -0,0 +1,313 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```

Comment:
I think this is:

* Add two fields `ShouldDrain` and `ShouldValidate`
* Call two (new) `drain` and `validate`

That's what it looked like the diff would be - if that's the case that is very acceptable risk IMO.